### PR TITLE
Issue 362: Move README quick start to vignette and improve general README formatting

### LINKED ISSUE #362: Moving README quick start to a vignette and slim down README using drop downs
At the moment the README is potentially fairly overwhelming to users and as the package functionality has expanded the quick start is increasingly not representative of what the package can actually do (as it just shows a nowcast). 

It might be sensible to move this quick start to its own vignette and to otherwise slim down the README using drop-down options so that users can opt in to additional information. 

There has been some discussion if a similar move for the `EpiNow2` package (see https://github.com/epiforecasts/EpiNow2/issues/482).

### PR DESCRIPTION
**Pull request by pearsonca**
## Description

This PR closes #362 .

It splits the README into a tighter, reorganized version and a vignette (for the quickstart code).

## Checklist

- [X] My PR is based on a package issue and I have explicitly linked it.
- [X] I have included the target issue or issues in the PR title in the for Issue(s) *issue-numbers*: PR title
- [X] I have read the [contribution guidelines](https://github.com/epinowcast/epinowcast/blob/main/CONTRIBUTING.md).
- [X] I have tested my changes locally.
- [X] My code follows the established coding standards.
- [x] I have added a news item linked to this PR.
- [ ] I have updated the package development version by one increment in both `NEWS.md` and the `DESCRIPTION`.
- [x] I have reviewed CI checks for this PR and addressed them as far as I am able.

### CODE REVIEW

**Comment by seabbs** [comment]
@jamesmbaazam would you be open to doing a firs review on this when its ready?

**Comment by jamesmbaazam** [comment]
@seabbs Sure. I'll do that now.

**Comment by seabbs** [comment]
@pearsonca you noted some frustration with the README CI in this PR. Could you open a discussion issue for deciding if and how to improve this part of the workflow?

**Comment by pearsonca** [comment]
@jamesmbaazam, for your review: please feel free to suggest in an edit to the `NEWS.md` to note your review.

@seabbs not sure if we've written down somewhere, but the above seems like the preferrable technique to capture reviewing in NEWS - PR author writes in the note, PR reviewers suggest in `reviewed by ...` revisions to record their contribution.

**Comment by pearsonca** [comment]
@seabbs re #378 - that pops currently in the newly-split-out vignette. I can amend this once that's fixed everywhere or just do it immediately (i *think* it won't conflict - just being erase from README, and new file is new file).

Added #379 re README CI.

File: vignettes/getting-started-part-I.Rmd
Comment (jamesmbaazam) [inline vignettes/getting-started-part-I.Rmd:100]: Because this is to "get started", I would normally prefer for the inputs to be assigned explicitly to the arguments. It's a nice way to introduce users to the argument names.

```suggestion
  r = ~ 0 + (1 | day), data = pobs
```

Comment (pearsonca) [reply vignettes/getting-started-part-I.Rmd:100]: seems reasonable to me; q to @seabbs - address this as part of improving the "getting started" guides generally or just do it now?

my general opinion for this amount / complexity: just do it here, but make a note of it as example of items to track down in detail as part of a PR dedicated to improving / expanding the guides.

File: vignettes/getting-started-part-I.Rmd
Comment (jamesmbaazam) [inline vignettes/getting-started-part-I.Rmd:113]:
```suggestion
reference_module <- enw_reference(parametric  = ~1, distribution = "lognormal", data = pobs)
```

Comment (jamesmbaazam) [reply vignettes/getting-started-part-I.Rmd:113]: Side note: What's the preferred style? First argument after the opening bracket, then others below? All on the same line like here, or first argument starts on a new line after the opening bracket (line in previous command above)?

File: vignettes/getting-started-part-I.Rmd
Comment (jamesmbaazam) [inline vignettes/getting-started-part-I.Rmd:201]: This is S3 though.

**Review by jamesmbaazam — DISMISSED** [review]
Thanks for tagging me @pearsonca. It all aligns with my thinking about what should go in a README vs Get Started Vignette.

I've left some line comments and made suggestions where possible.

An issue outside of the scope of this PR is why there is no legend for the plots. Is it a preferred style here?

**Comment by pearsonca** [comment]
> Thanks for taggin me @pearsonca. It all aligns with my thinking about what should go in a README vs Get Started Vignette.
> 
> I've left some line comments and made suggestions where possible.

Generally useful, thanks - there's a mild process question here as to whether to do these now or as their own thing. I think the overall size and scope is small-enough to be fine.
 
> An issue outside of the scope of this PR is why there is no legend for the plots. Is it a preferred style here?

It is an identified item: #323 - so maybe up next?

File: vignettes/getting-started-part-I.Rmd
Comment (seabbs) [reply vignettes/getting-started-part-I.Rmd:124]: I think we might want to put actual content changes in their own PR for the quick start vs here? That being said there are quite small?

**Comment by seabbs** [comment]
>  I can amend this once that's fixed everywhere

@pearsonca lets just fix it everywhere. Should have a PR today. 

> but the above seems like the preferrable technique to capture reviewing in NEWS - PR author writes in the note, PR reviewers suggest in reviewed by ... revisions to record their contribution.

I don't think this workflow is as well captured as it could/should be. We likely need a new issue to highlight this. 

> An issue outside of the scope of this PR is why there is no legend for the plots. Is it a preferred style here?

@jamesmbaazam I think there is an issue to address this (ah yes there is #323).

File: vignettes/getting-started-part-I.Rmd
Comment (seabbs) [reply vignettes/getting-started-part-I.Rmd:100]: Yes I agree this really lies in improving the quick start. @jamesmbaazam what do you think? If in agreement would you be happy to make a new issue listing these issues as part of a wider effort to improve the quick start vignette?

File: NEWS.md
Comment (seabbs) [inline NEWS.md:53]:
```suggestion
- Split README to focus on package-level issues and moved quickstart into a getting started vignette. See #375 by @pearsonca and reviewed by @jamesmbaazam and @seabbs.
```

File: vignettes/getting-started-part-I.Rmd
Comment (jamesmbaazam) [reply vignettes/getting-started-part-I.Rmd:124]: I think this PR already rewords some sections so these suggestions can fit in the changes being introduced here.

File: README.Rmd
Comment (seabbs) [inline README.Rmd:91]: Either in a new bullet or here we should explicitly mention the new quick start guides.

File: vignettes/getting-started-part-I.Rmd
Comment (seabbs) [inline vignettes/getting-started-part-I.Rmd:2]: Should we call this getting started nowcasting with epinowcast or similar  (to foreshadow other getting started options that may not be for serial consumption?

File: vignettes/getting-started-part-I.Rmd
Comment (seabbs) [inline vignettes/getting-started-part-I.Rmd:3]: Similarly should this reference nowcasting explicitly?

**Review by seabbs — APPROVED** [review]
This is great @pearsonca and good job reviewing @jamesmbaazam. I made a few suggestions which I think we should discuss and added a suggestion to the NEWs for @pearsonca to review

File: README.Rmd
Comment (pearsonca) [reply README.Rmd:91]: New bullet, I think; I'll figure out the pkgdown stuff and figure out the proper slug.

File: vignettes/getting-started-part-I.Rmd
Comment (pearsonca) [reply vignettes/getting-started-part-I.Rmd:2]: Very sensible, will update.

File: vignettes/getting-started-part-I.Rmd
Comment (pearsonca) [reply vignettes/getting-started-part-I.Rmd:100]: @jamesmbaazam if you're interested in taking this on as a more general improvement PR, recommend you create an issue for that, noting these classes of changes, then resolve these comments.

File: vignettes/getting-started-part-I.Rmd
Comment (pearsonca) [reply vignettes/getting-started-part-I.Rmd:124]: ibid re extracting this to general upgrade PR.

aside - I do like this style of change (e.g. cull advanced vs basic, "simple"). Additionally, I generally prefer voice to be addressed to the reader `For your research question, you might need to combine ...` - avoids elaborate passive voice constructions.

File: vignettes/getting-started-part-I.Rmd
Comment (pearsonca) [inline vignettes/getting-started-part-I.Rmd:2]:
```suggestion
title: "Getting Started with Epinowcast: Nowcasting"
```

File: vignettes/getting-started-part-I.Rmd
Comment (pearsonca) [inline vignettes/getting-started-part-I.Rmd:3]:
```suggestion
description: "A quick start example demonstrating use of epinowcast to nowcast hospital admissions."
```

File: vignettes/getting-started-part-I.Rmd
Comment (pearsonca) [reply vignettes/getting-started-part-I.Rmd:201]: agreed - i'm not familiar enough with this part of interface / package conception to know how to fix it. this should definitely go in a cleaning pass of this vignette.

File: vignettes/getting-started-part-I.Rmd
Comment (pearsonca) [inline vignettes/getting-started-part-I.Rmd:15]:
```suggestion
  %\VignetteIndexEntry{Getting Started with Epinowcast: Nowcasting}
```

File: README.Rmd
Comment (pearsonca) [inline README.Rmd:91]:
```suggestion
- [Package website](https://package.epinowcast.org/): This includes a function reference, model outline, and case studies using the package. The package site covers the release version, which can be installed from our Universe or from the latest GitHub release (see [installation instructions](#Installation)). Documentation for the development version (corresponding to the `main` branch on GitHub) [is also available](https://package.epinowcast.org/dev/).
- [Package Vignettes](https://package.epinowcast.org/articles): These provide tutorials, focused discussions of particular aspects, or demonstrate case studies. The [Getting Started with Epinowcast: Nowcasting](https://package.epinowcast.org/articles/getting-started-part-1) is a good place to start.
```

Comment (pearsonca) [reply README.Rmd:91]: amended

Comment (pearsonca) [reply README.Rmd:91]: aside: i was going to add a note that the vignettes are just available via the normal R interface, but when I checked, that was not the case? seems odd - perhaps I'm installing it wrong?

Comment (seabbs) [reply README.Rmd:91]: this is good but case studies are in both

File: vignettes/getting-started-part-I.Rmd
Comment (jamesmbaazam) [reply vignettes/getting-started-part-I.Rmd:124]: >  Additionally, I generally prefer voice to be addressed to the reader For your research question, you might need to combine ... - avoids elaborate passive voice constructions

I like this.

File: vignettes/getting-started-part-I.Rmd
Comment (jamesmbaazam) [reply vignettes/getting-started-part-I.Rmd:201]: I think it's a wording issue. Maybe the text is referring to using provided functions post `summary()`. Will note it down for upgrade.

File: vignettes/getting-started-part-I.Rmd
Comment (jamesmbaazam) [reply vignettes/getting-started-part-I.Rmd:100]: Done.

File: vignettes/getting-started-part-I.Rmd
Comment (jamesmbaazam) [reply vignettes/getting-started-part-I.Rmd:113]: I've linked an issue.

File: vignettes/getting-started-part-I.Rmd
Comment (jamesmbaazam) [reply vignettes/getting-started-part-I.Rmd:124]: See linked issue.

File: vignettes/getting-started-part-I.Rmd
Comment (jamesmbaazam) [reply vignettes/getting-started-part-I.Rmd:201]: See linked issue.

File: vignettes/getting-started-part-I.Rmd
Comment (pearsonca) [inline vignettes/getting-started-part-I.Rmd:44]:
```suggestion
  germany_covid19_hosp[location == "DE"][age_group == "00+"] |>
```

**Review by pearsonca — COMMENTED** [review]
amend per %in% to `==` transition

File: README.Rmd
Comment (pearsonca) [reply README.Rmd:91]: unclear - but clearly a separate issue if so. are the vignettes a CI build, perhaps?

File: vignettes/getting-started-part-I.Rmd.orig
Comment (pearsonca) [inline vignettes/getting-started-part-I.Rmd.orig:46]:
```suggestion
```

**Review by pearsonca — COMMENTED** [review]
Fixed one oversight re leftover troubleshooting code

**Comment by seabbs** [comment]
How are we getting along here? Nearly ready for another review? I see a loose Makefile is causing some problems.

**Comment by pearsonca** [comment]
> How are we getting along here? Nearly ready for another review? I see a loose Makefile is causing some problems.

Review time, methinks. Moved the make approach to a new issue for discussion.

File: _pkgdown.yml
Comment (seabbs) [inline _pkgdown.yml:33]: needs to be updated to the new name?

**Review by seabbs — DISMISSED** [review]
All looks very good but the new name for the vignette needs to be propagated into the pkgdown and the names of the figure outputs produced when rendering.

**Review by seabbs — APPROVED** [review]
Looks all good to me! I think this is a big improvement over what we had before.

**Comment by seabbs** [comment]
Thanks both for the work on this!

**Comment by jamesmbaazam** [comment]
Agreed. Sorry I've been away sick today. Do let me know if there's anything I can help with before merge.

**Comment by seabbs** [comment]
No worries at all - the first review is 90% of the battle. I hope you feeling better!

**Comment by jamesmbaazam** [comment]
Fighting with a nasty flu/cold (covid-negative) that started today 🤧.

**Comment by seabbs** [comment]
ick. Not ideal!

**Pull request merged**
